undefined reference to `__imp___acrt_iob_func'

The "under latest MinGW, cannot link with C code using stdout" issue was closed
today. I tried testing with this:

~~~
$ rustc -V
rustc 1.43.0-nightly (58b834344 2020-02-05)
~~~

https://github.com/rust-lang/rust/commit/58b834344

but I am still getting this type of result:

~~~
json.cpp:(.text.unlikely+0x34): undefined reference to `__imp___acrt_iob_func'
~~~

@mati865 ?

Using Cygwin with these packages:

~~~
curl
git
make
mingw64-x86_64-gcc-g++
~~~

I can compile a C library directly:

~~~
git clone git://github.com/sass/libsass
cd libsass
AR=x86_64-w64-mingw32-ar \
CC=x86_64-w64-mingw32-gcc \
CXX=x86_64-w64-mingw32-g++ \
make
~~~

but if I try the Rust crate:

~~~sh
git clone git://github.com/compass-rs/sass-rs
cd sass-rs
s1='Command::new("cc")'
s2='let r=env::var("CC").unwrap_or("cc".to_string());Command::new(\&r)'
sed -i "s/$s1/$s2/" sass-sys/build.rs
AR=x86_64-w64-mingw32-ar \
CC=x86_64-w64-mingw32-gcc \
CXX=x86_64-w64-mingw32-g++ \
../Rust/bin/cargo build
~~~

I get this:

~~~
json.cpp:(.text.unlikely+0x34): undefined reference to `__imp___acrt_iob_func'
~~~

I belive I'm getting the same error here for [x86_64 "coreutils" builds on GHA-CI](https://github.com/rivy/rust.coreutils/runs/432527747).

```
rustup 1.21.1 (7832b2ebe 2019-12-20)
stable-x86_64-pc-windows-msvc
nightly-x86_64-pc-windows-gnu (default)
cargo 1.42.0-nightly (9d32b7b01 2020-01-26)
rustc 1.43.0-nightly (442ae7f04 2020-02-06)
```
```
C:\Rust\.cargo\bin\cargo.exe build --release --target=x86_64-pc-windows-gnu --features windows
```
 =>
```
 = note: D:\a\rust.coreutils\rust.coreutils\target\x86_64-pc-windows-gnu\release\deps\libonig_sys-dde3d0c97310dd35.rlib(regexec.o):regexec.c:(.text$onig_setup_builtin_monitors_by_ascii_encoded_name+0xb7): undefined reference to `__imp___acrt_iob_func'`
```

(`gcc` call) ...
```
 = note: "gcc" "-fno-use-linker-plugin" "-Wl,--nxcompat" "-nostdlib" "-m64" "C:\\Rust\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\crt2.o" "C:\\Rust\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\rsbegin.o" "-L" "C:\\Rust\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "d:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\uutils-e75f1d0848693a1d.uutils.21wau9jq-cgu.0.rcgu.o" "d:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\uutils-e75f1d0848693a1d.uutils.21wau9jq-cgu.1.rcgu.o" "d:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\uutils-e75f1d0848693a1d.uutils.21wau9jq-cgu.10.rcgu.o" "d:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\uutils-e75f1d0848693a1d.uutils.21wau9jq-cgu.11.rcgu.o" "d:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\uutils-e75f1d0848693a1d.uutils.21wau9jq-cgu.12.rcgu.o" "d:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\uutils-e75f1d0848693a1d.uutils.21wau9jq-cgu.13.rcgu.o" "d:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\uutils-e75f1d0848693a1d.uutils.21wau9jq-cgu.14.rcgu.o" "d:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\uutils-e75f1d0848693a1d.uutils.21wau9jq-cgu.15.rcgu.o" "d:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\uutils-e75f1d0848693a1d.uutils.21wau9jq-cgu.2.rcgu.o" "d:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\uutils-e75f1d0848693a1d.uutils.21wau9jq-cgu.3.rcgu.o" "d:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\uutils-e75f1d0848693a1d.uutils.21wau9jq-cgu.4.rcgu.o" "d:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\uutils-e75f1d0848693a1d.uutils.21wau9jq-cgu.5.rcgu.o" "d:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\uutils-e75f1d0848693a1d.uutils.21wau9jq-cgu.6.rcgu.o" "d:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\uutils-e75f1d0848693a1d.uutils.21wau9jq-cgu.7.rcgu.o" "d:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\uutils-e75f1d0848693a1d.uutils.21wau9jq-cgu.8.rcgu.o" "d:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\uutils-e75f1d0848693a1d.uutils.21wau9jq-cgu.9.rcgu.o" "-o" "d:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\uutils-e75f1d0848693a1d.exe" "d:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\uutils-e75f1d0848693a1d.uovozkh7uukc4st.rcgu.o" "-Wl,--gc-sections" "-nodefaultlibs" "-L" "d:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps" "-L" "d:\\a\\rust.coreutils\\rust.coreutils\\target\\release\\deps" "-L" "C:\\Users\\runneradmin\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\winapi-x86_64-pc-windows-gnu-0.4.0\\lib" "-L" "d:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\build\\backtrace-sys-d112b5ac244f4e59\\out" "-L" "d:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\build\\onig_sys-9bd996fb8baf6f5e\\out" "-L" "C:\\Rust\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-Wl,-Bstatic" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_yes-64726386bd35fb5a.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_whoami-c12d24928295a4b0.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libadvapi32-7fde856c4f35643d.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_wc-c14e770d2ddc8655.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_uniq-ca2704529c8b2e33.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_unexpand-650982f58ae6e49f.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_tsort-c094b409e68277b9.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_truncate-89d62f37e9f86882.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_true-92f982f71c6c7a38.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_tr-3d4cd34a0627aeeb.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libfnv-6910f98a6c037b6e.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libbit_set-24850ed1e8898b30.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libbit_vec-61d31a0b0cb400bc.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_touch-f01a223a3527f5bd.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_test-18e18af8b44c476a.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_tee-2532f172cd9a62dd.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_tail-51fee45b957ba6a2.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_tac-7dceafc56ddbb27f.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_sync-02e6ab4532a43344.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_sum-40b9613997f59567.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_split-f96a8602c9e5dae7.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_sort-750d3dad19b4b3e3.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libsemver-6825aa73fef30634.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libsemver_parser-15ea0ae1edde55dd.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_sleep-6234104f93c0bd90.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_shuf-2e213bfeaf84d7c2.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_shred-29fd78332c577897.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_seq-8793b82c2f1f6e9e.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_rmdir-379d6dc5603975cd.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_rm-8a7a2e590ee8bf33.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libremove_dir_all-ada98e7d3175e6d1.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_relpath-0eaf3a852d978bee.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_realpath-a4e7b421d2bb4e30.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_readlink-453eacc4d6948cbf.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_pwd-7f4e77118b04211c.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_ptx-94e17de57bb15d39.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_printf-0c35757947a8b51e.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libitertools-8a97a3178da37835.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libeither-cd19299c0ea81528.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_printenv-9150c0c42c73e7d8.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_paste-288a3b84cd4ac2b8.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_od-915e2965db427dc2.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libhalf-70db04f6513ac415.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libbyteorder-383ae546194cfebf.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_nproc-a5f3ff40cd928be2.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libnum_cpus-afa7fda9c629938f.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_nl-f9e8706a0ee22848.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_mv-518e9761df617c7d.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_more-d536d4bbfcf14301.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_mktemp-6c9bc53c92298f44.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libtempfile-51ae08f43dad25f9.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\librand-58f4e3e1c8917246.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\librand-b1e9ba18fc7a1236.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_mkdir-0166e0f73c384200.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_ls-1bdce2049334f859.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libtermsize-690456f381136dce.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libterm_grid-0296d102cb7a0730.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libnumber_prefix-02a9ba2a7a72255c.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_ln-3e8c5f7afc84cc7e.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_link-5b83bdb37fe0c1e4.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_join-cbbb6a5b5d75d623.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_hostname-19670feb21ed3ec9.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_head-004a9c38dc8e69c4.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_hashsum-49bf332f4cdcd0eb.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libsha3-96c57bbc2538c869.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libsha2-22b1033778e0e761.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libfake_simd-84d227a564fe4098.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libblock_buffer-a5d612c5b748149c.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libdigest-db404003ca17a15a.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libbyte_tools-417f842dd1eaae7c.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libgeneric_array-52ca3f9673020713.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libnodrop-d75c2368426b0be9.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libtypenum-08e9af87054f9e2f.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libsha1-b05fb9c4256749ed.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libregex-10469cca662bc9ef.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libutf8_ranges-9a1f72e5f4c3603e.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libregex_syntax-ba04b5bacccaa250.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libucd_util-35511961900a9b26.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libthread_local-e3c17d173d6aa30d.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libaho_corasick-d079935f3fae365b.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libmemchr-5bf2481f6bd2e190.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libmd5-922b44a707b3fe76.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libhex-2edcebbf9aa8132f.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_fold-c0062693c0c11f98.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_fmt-f87d30aff96449f1.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_false-47e0fbc44f91310e.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_factor-d518dbf5f504ee71.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\librand-bf247390203c9446.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\librand_core-51891e94dfee7d60.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\librand_core-0ae0ee45d1a7f56c.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_expr-bbb6a4c294a6c64e.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libonig-5de97a0b77611950.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libonig_sys-dde3d0c97310dd35.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\liblazy_static-7839fb84c1bc2d12.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_expand-b8f4fbdab4d39319.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_env-48ac4b3d3c51d3cb.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libini-e258268f06e44050.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_echo-7706b4a1745d4b27.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_dirname-e4b1d067dd4df51a.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_dircolors-104e32eae7858eff.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libglob-92b36f243f6e1d5f.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_date-1e42ebdd02491293.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libchrono-5e2ec93383956488.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libnum_integer-6607992c689c94e9.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libnum_traits-aa774c8c9ac7f0a2.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libtime-837a8ab87b805819.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_cut-e8af18456d8135c3.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_cp-29987e4bd92d8151.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libkernel32-006f248cd6dec558.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libwinapi-b37b8c5936ddbbd1.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libwalkdir-a48d222f1f4a02cb.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libsame_file-2d1e71da32690796.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libwinapi_util-3259ae212fb6ecef.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libfiletime-e4cd383f89d149ab.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libclap-88a32b5114feb3c9.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libvec_map-499bf6e9c5df089a.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libtextwrap-62bcae8e5392f713.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libstrsim-8b137f737493d865.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libbitflags-d34869cdb4d2bca1.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libatty-93c11acbf5f89150.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_comm-bdbea6c93ca31fe3.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_cksum-b694b192bb979594.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_cat-2d447b73fa9559a4.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libquick_error-31091daf1ab9ee83.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_basename-f7d10dba2a9dbb59.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_base64-5f72fac38396bbf6.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_base32-9fdba7077e947a88.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuu_arch-b31e7fc211d28311.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libuucore-6c5ce856aa3cec31.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libdata_encoding-24a646431ba2c902.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libdunce-97958cbc2fcccef8.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libgetopts-28d336e36c5e2f4e.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libunicode_width-2ae19cb4e2d0fc3f.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libplatform_info-9c353671d4a68ae6.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libwinapi-6361f2254f70a9ab.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libnix-5dfda90bf8577bdb.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libfailure-2c9b4047291d5f86.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libbacktrace-7fc3ed8835d94898.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libbacktrace_sys-68e5c30c2dc36bac.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\librustc_demangle-c05db5a07ee03be6.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libcfg_if-909d0e8e77a73272.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\liblibc-fcc954dcda6d7b88.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libwild-069a1a054e059e72.rlib" "D:\\a\\rust.coreutils\\rust.coreutils\\target\\x86_64-pc-windows-gnu\\release\\deps\\libglob-31aa3d705b5e77e5.rlib" "-Wl,--start-group" "C:\\Rust\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libstd-27fabe47cbcb25db.rlib" "C:\\Rust\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libpanic_unwind-1feb757b8f479e3c.rlib" "C:\\Rust\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libhashbrown-ef7ba4a447ef95e5.rlib" "C:\\Rust\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\librustc_std_workspace_alloc-b99f7a67f773ea88.rlib" "C:\\Rust\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libbacktrace-c0157766cde324d6.rlib" "C:\\Rust\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libbacktrace_sys-53b9ca0566907b4b.rlib" "C:\\Rust\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\librustc_demangle-f671b94b301d2671.rlib" "C:\\Rust\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libunwind-2c24baa2e32f1e25.rlib" "C:\\Rust\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libcfg_if-8d7b111892c2c821.rlib" "C:\\Rust\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\liblibc-0e412f18ada22f56.rlib" "C:\\Rust\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\liballoc-86dcdc6cbb58680b.rlib" "C:\\Rust\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\librustc_std_workspace_core-da74c6c179536555.rlib" "C:\\Rust\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libcore-30b854551fc87f1e.rlib" "-Wl,--end-group" "C:\\Rust\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libcompiler_builtins-aeb4fe6bd4bc901a.rlib" "-Wl,-Bdynamic" "-ladvapi32" "-lkernel32" "-lwinapi_advapi32" "-lwinapi_cfgmgr32" "-lwinapi_credui" "-lwinapi_gdi32" "-lwinapi_kernel32" "-lwinapi_msimg32" "-lwinapi_opengl32" "-lwinapi_secur32" "-lwinapi_user32" "-lwinapi_version" "-lwinapi_winspool" "-lwinapi_ws2_32" "-ladvapi32" "-lws2_32" "-luserenv" "-Wl,-Bstatic" "-lgcc_eh" "-lpthread" "-Wl,-Bdynamic" "-lmingwex" "-lmingw32" "-lgcc" "-lmsvcrt" "-lmsvcrt" "-luser32" "-lkernel32" "C:\\Rust\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\rsend.o"
```

All other builds have been completing successfully for a long time (including the `i686-pc-windows-gnu` target).

Hmm, @mati865 ... I think this discussion is still directly on point. So, if you're happy to continue it here, I'll continue. If you'd like another issue opened, I can do that.

In summary, so far though:

With `gcc` ("gcc.exe (x86_64-posix-seh-rev0, Built by MinGW-W64 project) 8.1.0"; the 64-bit compiler, not multilib) on the path...

- i686-pc-windows-gnu => builds and tests successfully
- x86_64-pc-windows-gnu => fails with the `undefined reference to `__imp___acrt_iob_func'` error

When `gcc` is uninstalled ( via `choco uninstall mingw`), it leaves a perl `gcc` ("gcc.exe (x86_64-posix-seh, Built by strawberryperl.com project) 8.3.0"; the 64-bit compiler, not multilib) on the path. In that configuration, all builds/tests succeed.

I would think that the original configuration is exactly the case that the PR you just merged should have fixed.

Am I incorrect here?

Obviously, I can fix the build/test issue that I have in CICD by removing the offending `gcc`, but I'm happy to continue testing if it's helpful in creating a better overall solution. It might be nice, in the future, to be able to have both the usual `gcc` and rust usable for builds.

--- a/src/librustc_codegen_ssa/back/link.rs
+++ b/src/librustc_codegen_ssa/back/link.rs
@@ -1008,14 +1008,13 @@ fn get_crt_libs_path(sess: &Session) -> Option<PathBuf> {
                 path.pop();
                 path.pop();
                 // Based on Clang MinGW driver
-                let probe_path = path.join(&mingw_dir).join("lib");
-                if probe_path.exists() {
-                    return Some(probe_path);
-                };
-                let probe_path = path.join(&mingw_dir).join("sys-root/mingw/lib");
-                if probe_path.exists() {
-                    return Some(probe_path);
-                };
+                let probe_paths = vec!["lib", "sys-root/mingw/lib"];
+                for probe_path in probe_paths {
+                    let probe_path = path.join(&mingw_dir).join(&probe_path);
+                    if probe_path.join("crt2.o").exists() {
+                        return Some(probe_path);
+                    };
+                }
             };
         };
         None

--- a/src/librustc_codegen_ssa/back/link.rs
+++ b/src/librustc_codegen_ssa/back/link.rs
@@ -1002,15 +1002,22 @@ fn get_crt_libs_path(sess: &Session) -> Option<PathBuf> {
                     x if x == "x86" => "i686",
                     x => x,
                 };
+                let mingw_bits = &sess.target.target.target_pointer_width;
                 let mingw_dir = format!("{}-w64-mingw32", mingw_arch);
                 // Here we have path/bin/gcc but we need path/
                 let mut path = linker_path;
                 path.pop();
                 path.pop();
-                // Based on Clang MinGW driver
-                let probe_paths = vec!["lib", "sys-root/mingw/lib"];
+                // Loosely based on Clang MinGW driver
+                let probe_paths = vec![
+                    path.join(&mingw_dir).join("lib"),                // Typical path
+                    path.join(&mingw_dir).join("sys-root/mingw/lib"), // Rare path
+                    path.join(format!(
+                        "lib/mingw/tools/install/mingw{}/{}/lib",
+                        &mingw_bits, &mingw_dir
+                    )), // Chocolatey is creative
+                ];
                 for probe_path in probe_paths {
-                    let probe_path = path.join(&mingw_dir).join(&probe_path);
                     if probe_path.join("crt2.o").exists() {
                         return Some(probe_path);
                     };

